### PR TITLE
Extend Goals DB

### LINKED ISSUE #425: Extend the database and API of my goals to add sublists
// TODO: Add description

### PR DESCRIPTION
**Pull request by pulkitvyas08**
Resolves #425 
Part of #424, issue to be completed with @Tushar-4781's work on the issue, this PR has no dependency on that

NOTE: This PR will bring breaking changes to the indexdDB storage and interaction. We can discuss the implementation of a migration feature to avoid such changes in the future

### CODE REVIEW

File: src/models/GoalItem.ts [hunk 1]
```diff
@@ -4,12 +4,14 @@ export enum Repeat {
 }
 
 export interface GoalItem {
-  id?: number;
+  id?: Number;
   title: string;
   duration: Number;
-  sublist: string[] | null;
+  sublist?: GoalItem[];
```
Comment (Tushar-4781) [inline src/models/GoalItem.ts:10]: @pulkitvyas08 are you sure it should have GoalItem[] Interface?
@tijlleenders in the sublist is there something other than a string?  Acc to figma board like if I want to go shopping eve1y Wednesday then the sublist will have the items of it right?

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]
LGTM, just wanted confirmation on the below comment. will approve after that

File: src/models/GoalItem.ts (hunk 1 above)
Comment (pulkitvyas08) [reply src/models/GoalItem.ts:10]: Yes, refer to the attached snippet. All the sublist items can be deleted, archived, and have their title updated

image

Comment (Tushar-4781) [reply src/models/GoalItem.ts:10]: cool then. I think we need better example though because scenarios like above i dont see use of other attributes

Comment (pulkitvyas08) [reply src/models/GoalItem.ts:10]: The only one I see not being used is the plus icon. As sublist items cannot have their own sublist. Rest are good

Comment (tijlleenders) [reply src/models/GoalItem.ts:10]: The idea is that any list can have a subList, including subLists.
It's a Directed Acyclical Graph hierarchy that can go x levels deep.
That's why I propose separate goals, relationships and properties repositories.

**Pull request merged**